azurelinuxagent/common/utils/processutil.py

```diff
@@ -103,13 +103,24 @@ def _destroy_process(process, signal_to_send=signal.SIGKILL):
 def capture_from_process_modern(process, cmd, timeout):
     try:
         stdout, stderr = process.communicate(timeout=timeout)
+        # process completed
     except subprocess.TimeoutExpired:
-        # Just kill the process. The .communicate method will gather stdout/stderr, close those pipes, and reap
-        # the zombie process. That is, .communicate() does all the other stuff that _destroy_process does.
-        os.killpg(os.getpgid(process.pid), signal.SIGKILL)
-        stdout, stderr = process.communicate()
-        msg = format_stdout_stderr(sanitize(stdout), sanitize(stderr))
-        raise ExtensionError("Timeout({0}): {1}\n{2}".format(timeout, cmd, msg))
+        return_code = process.poll()
```

Comment:
I am surprised that this did not caused any tests to change.  Why?

Reply:
Because `process.poll()` will return `None` here, essentially matching the previous expected behavior. The one test we have does not fork, and tests that case. If we agree on this approach, I can add tests for the negative (forked) cases.

Reply:
I went ahead and added a new test to illustrate.